### ISSUE TITLE
runtime: TestNetpollDeadlock flakes across the board

### ISSUE THREAD

**Issue by FiloSottile** [golang/go#35353]
Saw it on a freebsd-amd64-12_0 TryBot, but it's been all over the place over the last few days.

```
--- FAIL: TestNetpollDeadlock (60.09s)
    crash_test.go:95: testprognet NetpollDeadlock exit status: exit status 2
    crash_test.go:450: output does not start with "done\n":
        dialing
        SIGQUIT: quit
        PC=0x45f1ae m=0 sigcode=65537
```

[2019-11-04T17:19:36-1da575a/linux-amd64](https://build.golang.org/log/f852eade059a09cf393080cc9f97128dd5653f2f)
[2019-11-04T15:18:34-d3660e8/linux-amd64-clang](https://build.golang.org/log/89d96c7102ff488d90ec66e6b92f61c9961fbc88)
[2019-11-04T10:13:30-1b2ff10/netbsd-amd64-8_0](https://build.golang.org/log/aed2f97eb97f7cd84ab9282cfb4011f015c41245)
[2019-11-03T04:58:01-32190b0/freebsd-amd64-race](https://build.golang.org/log/77b87085307b0899a3a58acb65e39b01eeed004b)
[2019-11-03T04:57:50-c6621d9/freebsd-amd64-race](https://build.golang.org/log/4fe705346c3e0c888cd4839f9fe4d47d95ed206f)
[2019-11-03T01:44:46-d2c039f/linux-amd64-nocgo](https://build.golang.org/log/a70879384993cff5dbf9b27d4f9a61c9cb7523e6)
[2019-11-02T21:51:16-d16ec13/linux-ppc64le-power9osu](https://build.golang.org/log/0c66a32666d56614b6d56cce27679f4a133a5e4e)
[2019-11-02T21:51:14-a3ffb0d/freebsd-arm64-dmgk](https://build.golang.org/log/c61029b8f727f5edea9f09bb1d0bfec8da9d3a24)
[2019-11-02T21:51:14-a3ffb0d/linux-arm64-packet](https://build.golang.org/log/315ea1702239a48c1e0ae1b9ac7df7f4573bd221)
[2019-11-02T19:25:54-afe50c1/linux-amd64](https://build.golang.org/log/f52e54c89d263e5d3d23ad0720c0c2d7d7178dbc)
[2019-11-02T00:36:59-48a0b97/openbsd-amd64-64](https://build.golang.org/log/5124e5cf7ce568db10367687dceb986610d3db78)
[2019-11-01T22:04:05-130f3c0/darwin-amd64-10_14](https://build.golang.org/log/4bb1328867d6050867910c5bb8afb082c66579ac)
[2019-11-01T20:16:01-c9d89f6/darwin-amd64-race](https://build.golang.org/log/978deb8a3dd3427dd6edc97ba52718b32bcfcd52)
[2019-11-01T14:45:24-c0555a2/freebsd-amd64-12_0](https://build.golang.org/log/992c7900b7b9aa0199718c0893c509a377c03ee6)
[2019-11-01T13:38:23-24e9c53/freebsd-amd64-race](https://build.golang.org/log/b468f0e6d7c9f2244c37064054b8f0eeb8c4cd8e)
[2019-11-01T05:38:51-e96fd13/freebsd-386-11_2](https://build.golang.org/log/ce52e3b6c3b1de14df903948ce972ac01dc18c62)
[2019-11-01T05:38:51-e96fd13/linux-amd64-race](https://build.golang.org/log/1f92dca26ea8fda8ad311dc7476839d93b7f3caa)
[2019-11-01T01:51:26-efd395f/freebsd-arm64-dmgk](https://build.golang.org/log/105329d0a1348fe535e5187083019a1fbf6120e2)
[2019-11-01T00:04:51-4a09a9b/linux-amd64-jessie](https://build.golang.org/log/b9fd22d82a3dbdc2c3773d2ba51ce7a808dd4852)
[2019-10-31T20:45:40-d6f2f83/freebsd-386-12_0](https://build.golang.org/log/6877cab1d3f8875bca409b75ca5d48171eba8b8e)
[2019-10-31T17:21:56-48c0cef/freebsd-386-11_2](https://build.golang.org/log/68d3b42a32a5146930195e356f8d91440d17bc51)
[2019-10-31T17:09:48-a9b37ae/linux-ppc64le-buildlet](https://build.golang.org/log/2388a87777c0a581bbc40f1c608de555a701d0d9)

/cc @ianlancetaylor

**Comment by FiloSottile** [comment]
Seems to have coincided with this rarer semi-related flake, which I noticed at https://storage.googleapis.com/go-build-log/b7ab0107/linux-amd64_99c6e062.log

```
runtime: netpollBreak write failed with 9
fatal error: runtime: netpollBreak write failed
```

[2019-11-04T16:32:38-7dcd343/darwin-amd64-nocgo](https://build.golang.org/log/b5379a4e31b2e1a85b263cc8f78e8b307c50e7a9)
[2019-11-02T21:51:21-177a36a/dragonfly-amd64](https://build.golang.org/log/5eaa9d78a17a052911bc2a89f015b162de0b9abc)
[2019-11-02T21:51:14-a3ffb0d/darwin-amd64-10_14](https://build.golang.org/log/bf86461e2f8e9f253211e904b133aa4105b56fbf)
[2019-11-02T21:51:07-40b7455/darwin-amd64-nocgo](https://build.golang.org/log/82f37cda0924b4f0dff98f6bcdb6cb0413d43cd7)
[2019-11-02T05:52:33-8de0bb7/netbsd-amd64-8_0](https://build.golang.org/log/bef33d0bbda8dc4623ee9e6af26f0e8117e0ee64)
[2019-11-01T21:41:41-9bde9b4/darwin-amd64-10_14](https://build.golang.org/log/9b4057bb5a0b522ffa67eefc13b0d74a6ca4ca13)
[2019-11-01T05:38:51-e96fd13/darwin-amd64-10_14](https://build.golang.org/log/3b971c4d9617f58bb6e738eb530c765148c0c18c)

**Comment by bradfitz** [comment]
Maybe https://go-review.googlesource.com/c/go/+/204800 fixes it.

**Comment by ianlancetaylor** [comment]
The `netpollBreak write failed with 9` failures should be fixed by https://golang.org/cl/204937, which is awaiting review.

I'm less sure about the `TestNetpollDeadlock` failures.  Let's see if they still happen now that CL 204800 is in.

**Comment by FiloSottile** [comment]
(FYI you did not assign a reviewer on CL 204937.)

**Comment by ianlancetaylor** [comment]
It wasn't in anyone's particular speciality.  I left it for the default approach.

**Comment by danscales** [comment]
I reproduced this on linux-amd64 at 1da575a7bc (the latest test failure above) by repeating this 100 times:

```
go test runtime -test.run TestNetpollDeadlock -test.count=100
```

(usually happened within about 20 times).  Using this reproduction case,  I showed that it was also fixed by Ian's commit d80ab3e85a  ("runtime: wake netpoller when dropping P, don't sleep too long in sysmon").  The failure is reproducible in commit immediate before Ian's, and not reproducible with Ian's commit.

So, I think we can close this bug as well.
